service-worker.js

```diff
@@ -208,26 +208,27 @@ function removeUrlParameters(url) {
 }
 
 /**
- * Looks up a Request in CACHE and returns a Promise for the matched Response
- * @param {Request} request The Request to fulfill from CACHE
- * @returns {Promise<Response>} A Promise for the cached Response 
+ * Looks up a Request in CACHE_NAME and returns a Promise for the matched Response
+ * @param {Request} request The Request to fulfill from CACHE_NAME
+ * @returns {Promise<Response>} A Promise for the cached Response
+ * @throws {Promise<String>} A rejected Promise with strings 'disabled' or 'no-match'
```

Comment:
I don't think we should use the `@throws` annotation for the case of a rejected Promise.
I suppose it is meant only for exceptions that are synchronously sent by the function (like in Java) : https://jsdoc.app/tags-throws.html

In this case, we might add this information (about what is rejected by the promise) as a comment at the end of the `@returns` line

Reply:
OK! Will remove before merging. It's somewhere in that long thread mentioned in #568, but you're right that it might not be the best way to do it, and it's not properly documented anywhere.